channeld/channeld.c

```diff
@@ -2381,36 +2390,58 @@ static void peer_reconnect(struct peer *peer,
 	}
 }
 
-/* Funding has locked in, and reached depth. */
-static void handle_funding_locked(struct peer *peer, const u8 *msg)
+/* ignores the funding_depth unless depth >= minimum_depth
+ * (except to update billboard, and set peer->depth_togo). */
+static void handle_funding_depth(struct peer *peer, const u8 *msg)
 {
-	unsigned int depth;
+	u32 depth;
+	struct short_channel_id *scid;
 
-	if (!fromwire_channel_funding_locked(msg,
-					     &peer->short_channel_ids[LOCAL],
-					     &depth))
-		master_badmsg(WIRE_CHANNEL_FUNDING_LOCKED, msg);
+	if (!fromwire_channel_funding_depth(tmpctx,
+					    msg,
+					    &scid,
+					    &depth))
+		master_badmsg(WIRE_CHANNEL_FUNDING_DEPTH, msg);
 
 	/* Too late, we're shutting down! */
 	if (peer->shutdown_sent[LOCAL])
 		return;
 
-	if (!peer->funding_locked[LOCAL]) {
-		status_trace("funding_locked: sending commit index %"PRIu64": %s",
-			     peer->next_index[LOCAL],
-			     type_to_string(tmpctx, struct pubkey,
-					    &peer->next_local_per_commit));
-		msg = towire_funding_locked(NULL,
-					    &peer->channel_id,
-					    &peer->next_local_per_commit);
-		sync_crypto_write(&peer->cs, PEER_FD, take(msg));
-		peer->funding_locked[LOCAL] = true;
-	}
+	if (depth < peer->channel->minimum_depth) {
+		if (peer->funding_locked[LOCAL] || peer->depth_togo == 0)
+			status_failed(STATUS_FAIL_INTERNAL_ERROR,
+					"funding shouldn't be locked");
 
-	peer->announce_depth_reached = (depth >= ANNOUNCE_MIN_DEPTH);
+		peer->depth_togo = peer->channel->minimum_depth - depth;
 
-	/* Send temporary or final announcements */
-	channel_announcement_negotiate(peer);
+	} else {
+
+		assert(scid);
```

Comment:
@SimonVrouwe  hi, Simon. what about this? I didn't remove else() and just placed bilboard_update() outside the block?

Reply:
Sorry I mixed up a few things (didn't see there was no return in the first  `if`), so please ignore my previous comment. Looks good now!

Reply:
Thank you:)